### ISSUE TITLE
Deprecated API: client.Databases.RetrieveAsync

### ISSUE THREAD

**Issue by matkoch** [notion-dotnet/notion-sdk-net#405]
**Describe the bug**
API was deprecated

**To Reproduce**
```
var notionClient = NotionClientFactory.Create(new ClientOptions
            {
                AuthToken = "accessToken"
            }
var database = await notionClient.Databases.RetrieveAsync("");
```

**Expected behavior**
Getting information about the database.

**Comment by KoditkarVedant** [comment]
@matkoch would you like to contribute? I would be happy to accept the PR.

**Comment by Kuzmanov** [comment]
Since it's not possible to get all databases from the databases endpoint, and it requires an `Id`, I think it makes sense to throw an `ArgumentNullException` when `databaseId` is not provided.

As an alternative, Notion suggests using the [Search API](https://developers.notion.com/reference/post-search)

**Comment by KoditkarVedant** [comment]
@Kuzmanov I agree, I will update the API to throw exception if missing DatabaseId as it is required field. @matkoch I don't think the `RetrieveAsync` is deprecated.

**Comment by KoditkarVedant** [comment]
closing this issue for now, fee free to re-open if still valid.
